entity-data-service-rx-client/src/main/java/org/hypertrace/entity/data/service/rxclient/EntityDataCachingClient.java

```diff
@@ -128,24 +122,20 @@ private void executeUpdate() {
       } finally {
         lock.unlock();
       }
-      Single<Entity> updateResult =
-          EntityDataCachingClient.this.createOrUpdateEntity(entityKey, condition).cache();
-
-      responseObservers.forEach(updateResult::subscribe);
+      EntityDataCachingClient.this
+          .createOrUpdateEntity(entityKey, condition)
+          .doOnError(error -> log.error("Error upserting entity", error))
```

Comment:
one downside of the void return is the error will never make it back - doesn't seem to me like it's worth creating a subscription per call though.

Reply:
Can we still throw the exception back, if there was an exception creating/updating the entity?

Reply:
no, we've taken off the return of this method, and since it's not blocking this code is executed long after it returns. Abstractly, I prefer the alternative, which I had working - returning a single. But practically, that means that we'll be maintaining a subscription for each call/span for that "eventually" duration (which is currently 15s), just to turn around and take this same action.

Reply:
This did remind me though, I need to break the habit of using `.doOnError` like this - since it doesn't catch it or subscribe to it. The logging still happens, but it'd be followed by a
```
io.reactivex.rxjava3.exceptions.UndeliverableException
The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling | 
```
The following commit illustrates how this is supposed to be handled.